Test/common.py

```diff
@@ -0,0 +1,15 @@
+import pytest
+import netket as nk
+
+skipif_mpi = pytest.mark.skipif(nk.utils.n_nodes > 1, reason="Only run without MPI")
+
+onlyif_mpi = pytest.mark.skipif(nk.utils.n_nodes < 2, reason="Only run with MPI")
+
+
+class one_rank:
+    def __enter__(self):
+        self._orig_nodes = nk.utils.n_nodes
+        nk.utils.n_nodes = 1
+
+    def __exit__(self, exc_type, exc_value, traceback):
+        nk.utils.n_nodes = self._orig_nodes
```

Comment:
Since this only sets `n_nodes`, what exactly is the idea behind this class and how is it meant to interact with `mpi4py` etc. which will use `COMM_WORLD.size` most likely? (Also, is it consistent to set `n_nodes` but not `netket.utils.rank`?

Reply:
This is a very ugly hack...
See how I use it [here](https://github.com/netket/netket/blob/e11c9837c84b8854004467e212737e1a6e7331e6/Test/Optimizer/test_sr_itersolve.py#L91).

It works because in netket, for performance and to avoid depending directly on mpi4py, we always have a case `if n_nodes == 1 ... else use_mpi` so this is effectively tricking netket into running the serial code.

Reply:
The idea is to check that algorithms give the same result with 1 node or under MPI.
unless we rewrite the MPI logic in netket I am not sure how we could avoid this.
Or we have to rewrite whole algorithms in the tests..

Reply:
Maybe for documentation purposes we should be a bit more verbose withe the name, maybe something like `netket_disable_mpi` (a bit like the `jax.api.disable_jit` context manager).

Reply:
done.